### ISSUE TITLE
EFCore 5: Database Collation Option Missing for Guid Properties

### ISSUE THREAD

**Issue by ab-tools** [msawczyn/EFDesigner#245]
Hello Michael,

just noticed one more thing:
It's great that we can now manually override the default database collation per property, but looks like this option is missing till now for `Guid` type properties.

As those are saved as `CHAR(36)` columns (at least for MySQL) I really would like to override the default collation `utf8mb4_0900_ai_ci` with `utf8mb4_bin` for performance reasons.

Is there a particular reason why this option is not available for `Guid` type properties despite those are saved as strings in the database (and thus have a column collation defined)?

Thanks
Andreas

**Comment by ab-tools** [comment]
Just to add to this, but not sure if this would be possible:
As there are obviously quite many identity properties in the model (all of type `Guid`) and we do have a `Default Identity Type` model property already (which I have set to `Guid` obviously), it would be great if there would be also a `Default Identity Collation` model property available in case the identity type is a `String` or `Guid`.

This would avoid the need of setting the collation for each identity property manually and potentially forgetting it.

**Comment by msawczyn** [comment]
Andreas,

You can set the default collation for the model level, as a property of the designer surface and, as you've seen, override that at the entity property level. 

Having the designer add collation statements for Guids, however, is a bit more problematic. Since the designer is database-agnostic, we have no way of knowing at design time how the database translation will implement the EF configuration, so we have to stay at the most generic level possible. That's done by only emitting collation statements for things listed as "string".

That being said, you can, in the context's partial class, add your own code to change the collation for a Guid property since you know that you'll be implementing with a database that stores Guids as strings. It's a manual step to be sure, but one of the advantages, I believe, of the designer that you can indeed to that kind of contextual customization.

I did find, however, that in 3.0.2 the code generation goes overboard with collation statements, now generating them for every property type if you set up a default at the model level and don't override it. We'll obviously need to pull back on that, since the migrator logic doesn't generate proper SQL when you specify a collation on an integer field :-(

Closing this as "won't do" for now. Should something different come up, please feel free to reopen or create another issue.

**Comment by ab-tools** [comment]
Thanks a lot, @msawczyn, for your detailed feedback, much appreciated!

I see, so you would basically just use the partial `OnModelCreatedImpl` method and then something like `modelBuilder.Entity<global::EntityTepy>().UseCollation("whatever")` for each entity you would want to change the collation?
That's definitely an option. I fully understand your point here, thanks for the explanations.

Regarding the default collation is there a need to add a collation statement to any entity (even strings) in this case at all?
I would expect that the default collation is just defined once for the whole database and then (in my case MySQL) is automatically applying this collation to every string column without explicitly setting that collation for that column. I expected that the general `modelBuilder.UseCollation("utf");` line is doing exactly that (defining a global collation for the whole database) or is that not the case?

Independent of that I'm currently a bit blocked by this strange issue here:
https://github.com/msawczyn/EFDesigner/issues/254
Same as the reporting user I'm currently unable to use 1-to-many relations as the collection stay empty for whatever reasons, but good to see that you're investigating this already!

Thanks again for all your support and happy New Year by the way :-)
Andreas
